src/Storages/MergeTree/MergeTreeData.h

```diff
@@ -923,6 +923,15 @@ class MergeTreeData : public IStorage, public WithMutableContext
 
     MergeTreePartsMover parts_mover;
 
+    /// Executors are common for both ReplicatedMergeTree and plain MergeTree
+    /// but they are being started and finished in derived classes, so let them be protected.
+    BackgroundJobsAssignee background_executor;
```

Comment:
We have `MergeTreeBackgroundExecutor` and `BackgroundJobsAssignee` but here we call Assignee `executor`. Very confusing.

Reply:
It was a piece of old code. Fixed.